j2k/src/org/jetbrains/kotlin/j2k/ExpressionConverter.kt

```diff
@@ -538,6 +538,13 @@ class DefaultExpressionConverter : JavaElementVisitor(), ExpressionConverter {
         val qualifier = expression.qualifierExpression
 
         var identifier = Identifier.withNoPrototype(referenceName, isNullable)
+
+
+        if (expression.getContainingClass()?.getParentOfType<PsiField>(false).let { it != null && it == expression.qualifierExpression?.reference?.resolve() }) {
```

Comment:
Why are you checking for `PsiField` specifically? Won't the same problem happen if the anonymous class is used as a local variable initializer?